crates/services/stf-runner/src/block.rs

```diff
@@ -0,0 +1,361 @@
+use sha2::{Digest, Sha256};
+use strata_asm_common::AsmLogEntry;
+use strata_primitives::{
+    buf::{Buf32, Buf64},
+    params::RollupParams,
+};
+
+use crate::account::{AccountId, SnarkAccountUpdate};
+
+/// Represents a complete block in the Orchestration Layer (OL) chain
+#[derive(Debug, Clone)]
+pub struct OLBlock {
+    signed_header: SignedOLBlockHeader,
+    body: OLBlockBody,
+}
+
+/// A block header with a cryptographic signature
+#[derive(Debug, Clone)]
+pub struct SignedOLBlockHeader {
+    header: OLBlockHeader,
+    signature: Buf64,
+}
+
+/// The header portion of an OL block containing metadata
+#[derive(Debug, Clone)]
+pub struct OLBlockHeader {
+    timestamp: u64,
+    slot: Slot,
+    epoch: Epoch,
+    parent_blockid: OLBlockId,
+    body_root: Buf32,
+    state_root: Buf32,
+}
+
+type OLBlockId = Buf32; // TODO: change this later
+type Slot = u64;
+type Epoch = u64;
+
+/// The body portion of an OL block containing the actual data
+#[derive(Debug, Clone)]
+pub struct OLBlockBody {
+    logs: Vec<OLLog>,
+    txs: Option<Vec<Transaction>>,
+    l1update: Option<L1Update>,
+}
+
+/// Represents a single transaction within a block
+// TODO: rename to OLTransaction?
+#[derive(Debug, Clone)]
+pub struct Transaction {
+    type_id: u16, // maybe this is not needed here since we use enum for payload? This seems to be
+    // relevant during serializing?
+    payload: TransactionPayload,
+    extra: TransactionExtra,
+}
+
+#[derive(Debug, Clone)]
+pub enum TransactionPayload {
+    GenericAccountMessage {
+        target: AccountId,
+        payload: Vec<u8>,
+    },
+    SnarkAccountUpdate {
+        target: AccountId, // is the transaction supposed to update the state of this target? looks
```

Comment:
Yes, that is what the spec says.

Reply:
Okay, so the name is confusing. This kind of reminds me of "upbound"/"downbound" conversation. :D